docs/rag/rag-initial-code-location.md

```diff
@@ -0,0 +1,109 @@
+# Code location for RAG
+
+| Created  | Dec 5, 2024 |
+| -------- | -------- |
+| Authors | Bill Murdock |
+| Replaces | N/A |
+| Replaced by | N/A |
+
+## What
+
+We want a retrieval-augmented generation (RAG) capability that provides outstanding results with minimal effort, is seamlessly integrated with InstructLab, and is also general enough to be used in other applications as well.
+
+## Why
+
+Many InstructLab users want to train a model and then use it to RAG.  Often they build something simple themselves for this purpose.  Two problems with this approach:
+
+- Building their own RAG is extra work.
+- Users who are not experts on RAG might not build a RAG that provides outstanding results.
+
+There is a very simple RAG capability at <https://github.com/instructlab/rag> .  It is not tightly integrated with InstructLab and it does not use any advanced RAG capabilities.  However, we have a directive from product management to not just unilaterally delete it or replace it with something radically different.
```

Comment:
should we be mentioning product management in a dev-doc purposed for an upstream project?

Reply:
The previous round of reviewers were explicitly requesting that any mention of this functionality having stakeholders that want it around be dropped because they named one stakeholder and said that this stakeholder was OK with it going away.  I was trying to balance the need to respond to that critique with enough information to convince them that there were still other stakeholders who do want it around without going so far as to name them.  It sounds like I went too far in the other direction.  I will adjust again.